### PR TITLE
Added kl_divergence.rs

### PR DESCRIPTION
**Pull request by ArvidHammarlund**
### Checklist

- [x] Confirm that `run-checks` script has been executed.

### Changes

Added kl-divergence loss for VAE

### Testing

Checked against a Numpy implementation.

### CODE REVIEW

**Review by nathanielsimard — COMMENTED** [review]
Thanks for contributing, It would be valuable to provide a PyTorch script to compare with, I'm not sure I can reproduce the results myself.

**Comment by ArvidHammarlund** [comment]
Pytorch only has the point-wise kl-divergence, which basically considers the difference in likelihood of two distributions seeing the same sample. This kl_divergence on the other hand considering the difference between the whole modelled probability distibutions (if both are normal).

The point wise kl_divergence has such as simple formula that I don't think it is necessary to make a module for it (as seen here: https://pytorch.org/docs/stable/generated/torch.nn.KLDivLoss.html). Additionally, the distribution kl_divergence is more applicable to VAE's which I believe is the main area of usage for KL_divegence? 

A reference a PyTorch code would be: https://github.com/AntixK/PyTorch-VAE/blob/master/models/vanilla_vae.py

**Comment by nathanielsimard** [comment]
@ArvidHammarlund This is really informative! I think it would be very valuable to have this as documentation over the module with the math notations. I'm going to work to make the latex rendering works pretty soon!
